perl/p5-dbd-mysql/Portfile

```
@@ -51,6 +51,13 @@ array set db_variants {
     mariadb10_4     mariadb-10.4
     percona         ""
 }
+# mariadb-10.2 does not build on Sequoia (15) and likely will not get
+# updated as it is end of life.  Add it only for OS versions earlier
+# than Sequoia for legacy support
+if {${os.major} < 15} {
```

Please check the OS name before checking its version, otherwise you may cause problems on non-Mac operating systems.

Also, on macOS, `${os.major}` is the Darwin version, not the macOS version. macOS 15 is Darwin 24.

```suggestion
if {${os.platform} eq "darwin" && ${os.major} < 24} {
```

---

Corrected in that last force push.  Thank you for the double check!